### PR TITLE
update circle ci yml: add execution on linux, macos, and windows

### PR DESCRIPTION
**Pull request by osy2**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,72 +1,125 @@
...
+        type: string
+    executor: << parameters.os >>
+    steps:
+      - checkout
+      - restore_cache:
+          key: notebook-data-${{ checksum 'Data/manifest.json' }}
+      - run:
+          name: Install Python << parameters.python-version >>
+          command: |
+            if [ "$RUNNER_OS" == "macos" ]; then
```
Comment (zivy) [inline .circleci/config.yml:58]: As the macos and linux share most of the code, possibly simplify as follows:
```
            if [ "$RUNNER_OS" == "macos" ]; then
              brew install pyenv
              eval "$(pyenv init -)"
            fi
            if [ "$RUNNER_OS" == "macos" || "$RUNNER_OS" == "linux" ]; then
              pyenv install << parameters.python-version >> || echo "Python version is already installed."
              pyenv global << parameters.python-version >>
              export PATH="$HOME/.pyenv/versions/<< parameters.python-version >>/bin:$PATH"
              echo -e '\neval "$(pyenv init -)"' >> ~/.bash_profile
              source ~/.bash_profile
            elif [ "$RUNNER_OS" == "windows" ]; then
              choco install python --version=<< parameters.python-version >> --allow-downgrade -y
              python -m pip install --upgrade pip
            fi
```

**Review by zivy — COMMENTED** [review]
Possibly simplify.

File: .circleci/config.yml (hunk 1 above)
Comment (osy2) [reply .circleci/config.yml:58]: Oh yeah. Thanks for that refactor.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,72 +1,121 @@
...
+          key: notebook-data-${{ checksum 'Data/manifest.json' }}
 workflows:
   test_notebooks:
     jobs:
       - lint
       - test:
+          matrix:
+            parameters:
+              python-version: ['3.7.6', '3.10.6']
+              os: ["linux", "macos", "windows"]
```
Comment (blowekamp) [inline .circleci/config.yml:113]: @zivy Is there really need to lint 6 times?

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,72 +1,121 @@
...
-      run: pytest -v --tb=short tests/test_notebooks.py::Test_notebooks::test_python_notebook
+      - checkout
+      - run:
+          name: Set up Python 3.9
+          command: |
+            sudo apt-get update
+            sudo apt-get install python3.9
+            sudo apt install python3.9-venv
+            python3.9 -m venv env
+            source env/bin/activate
```
Comment (blowekamp) [inline .circleci/config.yml:35]: Why not use the CircleCI image with python already installed:
https://circleci.com/docs/circleci-images/#python

And there is no need to use venv on the CI when the image is transient.

**Review by blowekamp — COMMENTED** [review]
Just some comments

**Pull request merged**